app/controllers/JudgmentNeutralCitationController.scala

```diff
@@ -0,0 +1,142 @@
+    val judgmentReference: Option[String] = if (effectiveNoNcnSelected) request.getQueryString(JUDGMENT_REFERENCE).filter(_.nonEmpty) else None
+    consignmentService
+      .getConsignmentRef(consignmentId, request.token.bearerAccessToken)
+      .map(reference => Ok(views.html.judgment.judgmentNeutralCitationNumber(consignmentId, reference, request.token.name, ncnValue, effectiveNoNcnSelected, judgmentReference)))
+  }
+
+  def validateNCN(consignmentId: UUID): Action[AnyContent] = judgmentUserAndTypeAction(consignmentId) { implicit request: Request[AnyContent] =>
+    val data: NeutralCitationData = extractFormData(request)
+    handleValidation(consignmentId, data)
+  }
+
+  private def handleValidation(consignmentId: UUID, ncnData: NeutralCitationData)(implicit request: Request[AnyContent]) = {
+
+    // Only send judgmentReference if noNeutralCitation is true
+    val judgmentReference: Option[String] = if (ncnData.noNeutralCitation) ncnData.judgmentReference else None
+
+    val validatedNCN = validateNCNDataValue(ncnData.neutralCitation, NCN).orElse(validateRelationships(ncnData))
+
+    if (validatedNCN.isDefined) {
+      consignmentService
+        .getConsignmentRef(consignmentId, request.token.bearerAccessToken)
+        .map(reference =>
+          BadRequest(
+            views.html.judgment.judgmentNeutralCitationNumberError(
+              consignmentId,
+              reference,
+              request.token.name,
+              validatedNCN.get,
+              ncnData.neutralCitation,
+              ncnData.noNeutralCitation,
+              judgmentReference
+            )
+          )
+        )
+    } else {
+      val validatedNoNCNReference = validateNCNDataValue(ncnData.judgmentReference, JUDGMENT_REFERENCE)
+      implicit val defaultLang: Lang = Lang.defaultLang
+      if (validatedNoNCNReference.isDefined) {
+        consignmentService
+          .getConsignmentRef(consignmentId, request.token.bearerAccessToken)
+          .map(reference =>
+            BadRequest(
+              views.html.judgment.judgmentNoNeutralCitationNumberReferenceError(
+                consignmentId,
+                reference,
+                request.token.name,
+                messages("SCHEMA_BASE.judgment_reference.maxLength"), // get this problem message from messages as not a validation error
+                ncnData.neutralCitation,
+                ncnData.noNeutralCitation,
+                judgmentReference,
+                validatedNoNCNReference
+              )
+            )
+          )
+      } else {
+        val url: String = buildBackURL(consignmentId, ncnData)
+        consignmentMetadataService
+          .addOrUpdateConsignmentNeutralCitationNumber(consignmentId, ncnData, request.token.bearerAccessToken)
+          .map(_ => Redirect(url))
+      }
+    }
+  }
+
+  private def buildBackURL(consignmentId: UUID, ncnData: NeutralCitationData) = {
```

Comment:
I think we don't need to build the back URL if we call the API to get the the data to show on the UI, we can simply redirect to upload page or from the upload page to NCN page